### PR TITLE
Support Uploading Bigger Files to Google Drive

### PR DESCRIPTION
**Pull request by ulsc**
Adding `resumable=True` parameter to `MediaFileUpload` gives the ability to upload files bigger than 5MB to Google Drive.

### CODE REVIEW

File: airflow/providers/google/suite/hooks/drive.py [hunk 1]
```diff
@@ -197,7 +197,7 @@ def upload_file(self, local_location: str, remote_location: str) -> str:
             parent = "root"
 
         file_metadata = {"name": file_name, "parents": [parent]}
-        media = MediaFileUpload(local_location)
+        media = MediaFileUpload(local_location, resumable=True)
```
Comment (potiuk) [inline airflow/providers/google/suite/hooks/drive.py]: Any drawbacks for using resumable? Should it be turned into a parameter maybe?

Comment (ulsc) [reply airflow/providers/google/suite/hooks/drive.py]: afaik it has no drawbacks. But we can use a parameter, of course.

Comment (potiuk) [reply airflow/providers/google/suite/hooks/drive.py]: Quick look and at least one issue https://github.com/googleapis/google-api-python-client/issues/625 which has been closed by inactivity. I think it **may** change behaviour, so having a parameter (defaulting to False) is much better.

Also that means some unit test shoudl be added.

Comment (potiuk) [reply airflow/providers/google/suite/hooks/drive.py]: Also - from that thread - it sems advisable that we could override the chunk_size (so maybe a close look at what parameters to expose might make sense).

**Comment by ulsc** [comment]
Added chunk_size & resumable as parameters.

**Comment by potiuk** [comment]
Right. I think unit tests cases in this case might be a bit over-the top :)

**Comment by ulsc** [comment]
I'm preparing to push the Google Drive Operators I use, will probably cover these arguments in the test cases.

File: airflow/providers/google/suite/hooks/drive.py [hunk 2]
```diff
@@ -180,12 +180,25 @@ def get_file_id(self, folder_id: str, file_name: str, drive_id: Optional[str] =
             file_metadata = {"id": files['files'][0]['id'], "mime_type": files['files'][0]['mimeType']}
         return file_metadata
 
-    def upload_file(self, local_location: str, remote_location: str) -> str:
+    def upload_file(
+        self,
+        local_location: str,
+        remote_location: str,
+        chunk_size: int = 104857600,
```
Comment (uranusjr) [inline airflow/providers/google/suite/hooks/drive.py]: Why this magic number?

Comment (ulsc) [reply airflow/providers/google/suite/hooks/drive.py]: It's the default chunk size from the original MediaFileUpload class: 100 x 1024 x 1024

Comment (uranusjr) [reply airflow/providers/google/suite/hooks/drive.py]: Thanks. Maybe add a comment explaining this?

Comment (ulsc) [reply airflow/providers/google/suite/hooks/drive.py]: I'm trying to understand this part also: 

> :param chunk_size: File will be uploaded in chunks of this many bytes. Only
            used if resumable=True. Pass in a value of -1 if the file is to be
            uploaded as a single chunk. Note that Google App Engine has a 5MB limit
            on request size, so you should never set your chunksize larger than 5MB,
            or to -1.

If the chunk size shouldn't be larger than 5MB, why would they put 100MB as default?

Comment (potiuk) [reply airflow/providers/google/suite/hooks/drive.py]: > If the chunk size shouldn't be larger than 5MB, why would they put 100MB as default?

It's only Google App Engine, not Google Drive.

Comment (potiuk) [reply airflow/providers/google/suite/hooks/drive.py]: > Thanks. Maybe add a comment explaining this?

Write it as `100*1024*1024` please too .

Comment (ulsc) [reply airflow/providers/google/suite/hooks/drive.py]: It's written like this in the hook's `download_file` method. I think we should keep it like this and change all of them afterwards if necessary.

Comment (potiuk) [reply airflow/providers/google/suite/hooks/drive.py]: NA... Change it straight away - it make more sense to explicitly show where it is from. And parser will anyhow optimize it and calculate.

BTW. Also "chunksize" is not liked by Sphinx it seems :). The  `chunk size` would be better.

Comment (ulsc) [reply airflow/providers/google/suite/hooks/drive.py]: done

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Cool!

**Pull request merged**